app/uk/gov/hmrc/economiccrimelevyregistration/controllers/deregister/DeregisterCheckYourAnswersController.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.controllers.deregister
+
+import play.api.i18n.{I18nSupport, Messages}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.economiccrimelevyregistration.config.AppConfig
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.actions.AuthorisedActionWithEnrolmentCheck
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.actions.deregister.DeregistrationDataRetrievalAction
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.{BaseController, ErrorHandler}
+import uk.gov.hmrc.economiccrimelevyregistration.models.ContactDetails
+import uk.gov.hmrc.economiccrimelevyregistration.models.deregister.Deregistration
+import uk.gov.hmrc.economiccrimelevyregistration.services.EclRegistrationService
+import uk.gov.hmrc.economiccrimelevyregistration.services.deregister.DeregistrationService
+import uk.gov.hmrc.economiccrimelevyregistration.viewmodels.deregister._
+import uk.gov.hmrc.economiccrimelevyregistration.viewmodels.govuk.SummaryListFluency
+import uk.gov.hmrc.economiccrimelevyregistration.views.html.ErrorTemplate
+import uk.gov.hmrc.economiccrimelevyregistration.views.html.deregister.DeregisterCheckYourAnswersView
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryList
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class DeregisterCheckYourAnswersController @Inject() (
+  val controllerComponents: MessagesControllerComponents,
+  authorise: AuthorisedActionWithEnrolmentCheck,
+  getDeregistrationData: DeregistrationDataRetrievalAction,
+  eclRegistrationService: EclRegistrationService,
+  deregistrationService: DeregistrationService,
+  appConfig: AppConfig,
+  view: DeregisterCheckYourAnswersView
+)(implicit ec: ExecutionContext, errorTemplate: ErrorTemplate)
+    extends FrontendBaseController
+    with I18nSupport
+    with BaseController
+    with ErrorHandler
+    with SummaryListFluency {
+
+  def onPageLoad(): Action[AnyContent] = (authorise andThen getDeregistrationData).async { implicit request =>
+    val eclReference = request.eclRegistrationReference.getOrElse("")
```

Comment:
is it okay not to have an eclReference here or should we return an error in this case?

Reply:
If you have no ECL reference then the `getSubscription` will fail so an error will be returned.

Reply:
We shouldn't be using getOrElse here.

Match on request.eclRegistrationReference and handle the error case.